Purchases/Purchasing/OfferingsManager.swift

```diff
@@ -115,9 +115,7 @@ private extension OfferingsManager {
                 return
             }
 
-            let productsByID = products.reduce(into: [:]) { result, product in
-                result[product.productIdentifier] = product
-            }
+            let productsByID = Dictionary(uniqueKeysWithValues: products.map { ($0.productIdentifier, $0) })
```

Comment:
Small refactor. `products` was a set so the uniqueness is guaranteed, therefore these 2 are equivalent.